### DISCUSSION TITLE
Dynamic Routing in NextJs is not working

### DISCUSSION THREAD

**Discussion by MuzmmilKhan1** [vercel/next.js#51934] [App Router]
I copied the whole code From the git repository: https://github.com/vercel/next-learn/tree/master/basics/dynamic-routes-step-1                 You can see my code here: https://github.com/MuzmmilKhan1/NextJsIssue                                                                                                          Can you please see and tell me what could be the possible issue? 
Thanks.

**Comment by icyJoseph** [comment]
What problem are you seeing? I mean, what are you trying to do that "is not working"?

**Comment by icyJoseph** [comment]
@MuzmmilKhan1 I see one issue already, you have `list/post.js`, but the helper file needed should be `list/posts.js` - 

Here, https://github.com/MuzmmilKhan1/NextJsIssue/blob/main/pages/post/%5Bid%5D.js#L2 you do:

```js
import { getAllPostIds, getPostData } from '../../lib/posts'
```

But that file doesn't exist. Instead you have: `lib/post.js`

**Reply by MuzmmilKhan1** [reply]
Ohh Thanks Mate It works Now. Thank you very much

**Reply by icyJoseph** [reply]
Don't forget to mark it as answered and close it :)

**Reply by mesuhailpm** [reply]
He left it as unanswered

**Comment by MuzmmilKhan1** [comment]
> @MuzmmilKhan1 I see one issue already, you have `list/post.js`, but the helper file needed should be `list/posts.js` -
> 
> Here, https://github.com/MuzmmilKhan1/NextJsIssue/blob/main/pages/post/%5Bid%5D.js#L2 you do:
> 
> ```js
> import { getAllPostIds, getPostData } from '../../lib/posts'
> ```
> 
> But that file doesn't exist. Instead you have: `lib/post.js`
